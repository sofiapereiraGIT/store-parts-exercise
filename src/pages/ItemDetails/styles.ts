import styled from "styled-components";

export const ItemDetails = {
  PageWrapper: styled.div`
    flex: 0 50em;
    height: fit-content;
  `,
};
