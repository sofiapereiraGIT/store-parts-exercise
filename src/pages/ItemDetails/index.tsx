import ItemDetailsList from "../../components/ItemDetailsList";
import { ItemDetails as S } from "./styles";

export default function ItemDetails() {
  return (
    <S.PageWrapper>
      <h2>Item Details</h2>
      <ItemDetailsList />
    </S.PageWrapper>
  );
}
