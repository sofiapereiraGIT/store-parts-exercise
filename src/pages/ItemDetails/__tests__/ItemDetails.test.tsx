import React from "react";
import renderer from "react-test-renderer";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import ItemDetails from "../index";

describe("ItemDetails Tests", () => {
  it("Renders properly", () => {
    const history = createMemoryHistory({
      initialEntries: [
        {
          state: {
            name: "itemName",
            type: "itemType",
            price: "itemPrice",
          },
        },
      ],
    });
    const renderedComponent = renderer.create(
      <Router history={history}>
        <ItemDetails />
      </Router>
    );
    expect(renderedComponent.toJSON()).toMatchSnapshot();
  });
});
