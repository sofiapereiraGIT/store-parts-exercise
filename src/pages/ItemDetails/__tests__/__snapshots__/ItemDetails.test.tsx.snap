// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails Tests Renders properly 1`] = `
<div
  className="sc-hKMtZM jFgHfc"
>
  <h2>
    Item Details
  </h2>
  <div
    className="sc-bczRLJ gtXPJG"
  >
    <div
      className="sc-gsnTZi eDBMpS"
    >
      <span
        className="sc-dkzDqf iBjklj"
      >
        name
        :
      </span>
       
      <span>
        itemName
      </span>
    </div>
    <div
      className="sc-gsnTZi eDBMpS"
    >
      <span
        className="sc-dkzDqf iBjklj"
      >
        type
        :
      </span>
       
      <span>
        itemType
      </span>
    </div>
    <div
      className="sc-gsnTZi eDBMpS"
    >
      <span
        className="sc-dkzDqf iBjklj"
      >
        price
        :
      </span>
       
      <span>
        itemPrice
      </span>
    </div>
  </div>
</div>
`;
